src/components/ForgotPassword/ForgotPassword.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useRef, useState } from 'react';
+
+import './ForgotPassword.css';
+import { useCountdown } from '../../hooks/useCountDown.ts';
+import Button from '../Button/Button.tsx';
+import ClockIcon from '../Icons/ClockIcon.tsx';
+
+// TODO: Investigate if we're able to use heartbeat to check if forgot password request already in progress and how the page react if there is one
```

Comment:
To answer that: yes, we can add a check to the profile status for the password reset token, if it exists, we return that the user is in the "reset password" state.